### PR TITLE
Document config owner

### LINKED ISSUE #694: cash.Configuration.Owner is not used
~[`cash.Configuration.Owner`](https://github.com/iov-one/weave/blob/master/x/cash/codec.proto#L40-L45) attribute seems to not be used~.

~Either remove it or~ add documentation of what this field is used for.

### PR DESCRIPTION
**Pull request by ethanfrey**
Closes #694 

Also added Metadata fields to the cash.Configuration structs to resolve the TODOs. The Msg Metadata is required by cash.NewConfigHandler() already, due to use in `SchemaMigratingRegistry`. The Configuratin Metadata is not used yet, but is good to keep track of changes (missing Metadata is same as schema 1, and I think we have to handle this manually, as gconf cannot import migrations)

### CODE REVIEW

**Comment by ruseinov** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry. <br />
You can find it at <a href="https://github.com/iov-one/weave/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ Breaking changes
 
 - messages produced by `cmd/bnscli` have a new binary format incompatible with
   the previous version.
+- cash.UpdateConfigurationMsg requires Metadata.Schema
```
Comment (husio) [inline CHANGELOG.md]: You can use ` (backtick) to apply code formatting and make both first and last word look nicer

File: spec/gogo/x/cash/codec.proto [hunk 2]
```diff
@@ -37,13 +37,16 @@ message FeeInfo {
 }
 
 message Configuration {
-  // TODO: add schema uint32 here
+  weave.Metadata metadata = 1;
+  // Owner is present to implement gconf.OwnedConfig interface
+  // This defines the Address that is allowed to update the Configuration object and is
+  // needed to make use of gconf.NewUpdateConfigurationHandler
```
Comment (husio) [inline spec/gogo/x/cash/codec.proto]: This is gold! :100: 

It took me quite some time to figure this out again, once I forgot that there is a `GetOwner` method expectation that is used by an interface.

File: spec/gogo/x/cash/codec.proto [hunk 3]
```diff
@@ -37,13 +37,16 @@ message FeeInfo {
...
   bytes owner = 2 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
   bytes collector_address = 3 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
   coin.Coin minimal_fee = 4 [(gogoproto.nullable) = false];
 }
 
 message ConfigurationMsg {
-  // TODO: add schema uint32 here
+  weave.Metadata metadata = 1;
```
Comment (husio) [inline spec/gogo/x/cash/codec.proto]: Now that both `Configuration` model and `ConfigurationMsg` message include schema, they must be registered with migration
- https://github.com/iov-one/weave/blob/master/x/cash/msg.go#L11
- https://github.com/iov-one/weave/blob/master/x/cash/model.go#L13

**Pull request merged**